packages/editor/src/components/post-title/index.js

```diff
@@ -36,12 +36,21 @@ class PostTitle extends Component {
 		this.onUnselect = this.onUnselect.bind( this );
 		this.onKeyDown = this.onKeyDown.bind( this );
 		this.redirectHistory = this.redirectHistory.bind( this );
+		this.textareaRef = createRef();
 
 		this.state = {
 			isSelected: false,
 		};
 	}
 
+	componentDidMount() {
+		// If there is a post title and it's empty, we should focus the title so
+		// the user can start typing the title right away.
+		if ( ! this.props.title || ! this.props.title.length ) {
+			this.textareaRef.current.textarea.focus();
```

Comment:
This will have another maybe "unwanted" side effect? If you switch to code mode and you switch back to editor without touching the title, it will be autofocused?